src/hooks.js

```diff
@@ -17,26 +17,17 @@ export function useEffectAfterMount(cb, dependencies) {
   }, dependencies);
 }
 
-function useEffectOnMount(cb) {
-  const justMounted = useRef(true);
-  useEffect(() => {
-    if (justMounted.current) {
-      return cb();
-    }
-    justMounted.current = false;
-  });
-}
-
+// Unique ID implementation borrowed from React UI :)
+// https://github.com/reach/reach-ui/blob/6e9dbcf716d5c9a3420e062e5bac1ac4671d01cb/packages/auto-id/src/index.js
 let idCounter = 0;
+const genId = () => ++idCounter;
 
 /**
  * This generates a unique ID for an instance of Collapse
  * @return {String} the unique ID
  */
 export function useUniqueId() {
-  let counter = useMemo(() => idCounter++, []);
-  useEffectOnMount(() => {
-    counter++;
-  });
-  return counter;
+  const [id, setId] = useState(null);
```

Comment:
Hi there, thanks for raising this. This was a patch since it fixes an issue with the internal behavior of the component. I do not recommend snapshotting internals of vendor libraries in your tests, since that snapshot would be testing _this_ library, not yours 😉

That said, it's odd that the ID is initializing with `null` and not `1` like I would expect. Can you share the code for your test?

Reply:
I can't share it sorry - I dont have rights to do so. But its not rly important for this case: we simply get props for button from useCollapse method and then we use these props to render component. So its not directly testing _this_ lib but testing our component with props from _this_ lib.

Now, when useCollapse returns null in id until didMount, the shallow snapshot of any component would be diffrent if useCollapse is called in constructor (or anywhere else before didMount or didUpdate lifecycle method).

We just updated snapshots so its not any bug or issue for us but we had interesting discussion in our team about this and I was just curious why to init this id exactly with _null_ when its not (or is it?) important for solving SSR id math issue.